frontend/src/metabase-lib/expressions/typeinferencer.js

```diff
@@ -6,6 +6,7 @@ export const MONOTYPE = {
   Number: "number",
   String: "string",
   Boolean: "boolean",
+  DateTime: "datetime",
```

Comment:
Adding a new type is usually very tricky. I suggest adding some more extensive tests to ensure assorted type conversion corner cases.

Reply:
Can you point to an example of a test that you had in mind? That would be very helpful.

Reply:
@ariya Are these tests sufficient? https://github.com/metabase/metabase/commit/50b06e97aec68c612221a3cc826ee5323b3fc7a8